pkg/controller/volume/attachdetach/attach_detach_controller_test.go

```diff
@@ -393,7 +415,17 @@ func Test_ADC_VolumeAttachmentRecovery(t *testing.T) {
 func volumeAttachmentRecoveryTestCase(t *testing.T, tc vaTest) {
 	fakeKubeClient := controllervolumetesting.CreateTestClient()
 	informerFactory := informers.NewSharedInformerFactory(fakeKubeClient, time.Second*1)
-	plugins := controllervolumetesting.CreateTestPlugin()
+	var plugins []volume.VolumePlugin
+	if tc.csiMigration {
+		defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIMigration, tc.csiMigration)()
+		defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIMigrationGCE, tc.csiMigration)()
+		defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.InTreePluginGCEUnregister, tc.csiMigration)()
```

Comment:
InTreePluginGCEUnregister does not exist in 1.18. You may want to change it to CSIMigrationGCEComplete.